metaflow/plugins/aws/step_functions/dynamo_db_client.py

```diff
@@ -65,6 +66,16 @@ def get_parent_task_ids_for_foreach_join(self,
         return response['Item']['parent_task_ids_for_foreach_join']['SS']
 
     def _get_instance_region(self):
-        return os.popen(
-            'curl -s http://169.254.169.254/latest/meta-data/placement/availability-zone/'
-            ).read()[:-1]
\ No newline at end of file
+        try:
+            r = requests.post(
+                url = "http://169.254.169.254/latest/meta-data/placement/availability-zone/"
+            )
+            if r.status_code != 200:
+                raise RuntimeError("Failed to query AWS region from " +
+                        "http://169.254.169.254/latest/meta-data/placement/availability-zone/\n" +
+                        "Error code: " + str(r.status_code))
+            return r.text[:-1]
+        except Exception as e:
```

Comment:
We might not need this block. All uncaught exceptions are caught in the task runtime, printed and re-raised appropriately.